### PR TITLE
Tempo: TraceQL metrics step option

### PR DESCRIPTION
**Pull request by adrapereira**
**What is this feature?**

This PR introduces an option to set the step for TraceQL metrics queries for the Search and TraceQL tabs.

**Why do we need this feature?**

Tempo sets a default step based on the time range but it can be useful to override the step for a query. This is also possible in Prometheus for example. 

**Who is this feature for?**

Users of the Tempo datasource. 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by joey-grafana — COMMENTED** [review]
Few comments but overall looks good 👍 

Can you link me the Tempo docs/PR for compare/minStep?

We should also add `minStep` to the docs. I've noticed that [the docs](https://grafana.com/docs/grafana/latest/datasources/tempo/query-editor/) are missing the span limit and table format options and several images are also out of date that do not show these options (for example, [here](https://grafana.com/docs/grafana/latest/datasources/tempo/query-editor/traceql-search/)).

This can be done in a subsequent PR, @knylander-grafana is this something you'd be interested in collaborating on?

File: public/app/plugins/datasource/tempo/traceql/TempoQueryBuilderOptions.tsx [hunk 1]
```diff
@@ -87,6 +91,20 @@ export const TempoQueryBuilderOptions = React.memo<Props>(({ onChange, query })
...
+          <EditorField
+            label="Step"
+            tooltip="Defines the step for metric queries. Use duration notation, for example 30s or 1m"
+          >
+            <AutoSizeInput
+              className="width-4"
+              placeholder="auto"
+              type="string"
+              min={1}
+              defaultValue={query.step}
```
[the remark this answers is not in the document]
Comment (joey-grafana) [reply public/app/plugins/datasource/tempo/traceql/TempoQueryBuilderOptions.tsx:102]: `auto` should not be the placeholder in that case. It should be something like `30s` or even better the actual value that Tempo uses as the computed step (if we know this as there is no docs available) if we don't send a `step` so users know what the `step` will default to if they don't enter a value.

In any case, we should only set the `step` in the request if we actually have a value to send.

We could even take this a step further and validate the `step` value but we can leave that for another PR.

[the remark this answers is not in the document]
Comment (joey-grafana) [reply public/app/plugins/datasource/tempo/traceql/TempoQueryBuilderOptions.tsx:102]: Yeah agreed we're better off not replicating that code. Also, we're better off in this case doing the same as Prometheus and we can talk about updating that later if necessary.

**Review by joey-grafana — APPROVED** [review]

**Comment by knylander-grafana** [comment]
> the docs
@joey-grafana I have added this content to my doc issue to update the Query editor: https://github.com/grafana/tempo-squad/issues/339

**Pull request merged**